src/OrbitBase/StopToken.cpp

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "OrbitBase/StopToken.h"
+
+#include "OrbitBase/Logging.h"
+
+namespace orbit_base {
+
+bool StopToken::IsStopRequested() const {
+  ORBIT_CHECK(IsStopPossible());
+  absl::MutexLock lock(&shared_stop_state_->mutex);
+  return shared_stop_state_->IsFinished();
+}
```

Comment:
Nit: I would consider defining this inline.

Reply:
done. Also moved the 2 StopSource methods into the StopSource header